core-web/libs/edit-content/src/lib/utils/functions.util.ts

```diff
@@ -293,3 +294,28 @@ export const transformFormDataFn = (contentType: DotCMSContentType): Tab[] => {
         }))
     }));
 };
+
+/**
+ * Generates a preview URL for a given contentlet.
+ *
+ * @param {DotCMSContentlet} contentlet - The contentlet object containing the necessary data.
+ * @returns {string} The generated preview URL.
+ */
+export const generatePreviewUrl = (contentlet: DotCMSContentlet): string => {
+    if (
+        !contentlet.URL_MAP_FOR_CONTENT ||
+        !contentlet.host ||
+        contentlet.languageId === undefined
+    ) {
+        console.warn('Missing required contentlet attributes to generate preview URL');
+
+        return '';
+    }
+
+    const baseUrl = `${window.location.origin}/dotAdmin/#/edit-page/content`;
+    const url = encodeURIComponent(contentlet.URL_MAP_FOR_CONTENT);
+    const hostId = contentlet.host;
+    const languageId = contentlet.languageId;
+
+    return `${baseUrl}?url=${url}%3Fhost_id%3D${hostId}&language_id=${languageId}&com.dotmarketing.persona.id=modes.persona.no.persona&editorMode=edit`;
```

Comment:
You can use the [URLSearchParams API](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams) to do this